tests/CMakeLists.txt

```diff
@@ -65,16 +65,29 @@ foreach(source_file ${source_files})
         ${ARGP_INCLUDEDIR}
         ${STATGRAB_INCLUDEDIR}
         ${LZ4_INCLUDEDIR}
+        ${BZIP2_INCLUDEDIR}
+        ${ZLIB_INCLUDEDIR}
         ${STC_DIR}/include
         ${CMAKE_SOURCE_DIR}/include
         ${CMAKE_SOURCE_DIR}/src
         ${CMAKE_CURRENT_SOURCE_DIR}/include
         ${CMAKE_CURRENT_BINARY_DIR}/include
     )
+
+    # Linking tests to libasdf does not work. Some symbols used by the test(s) are marked "hidden"
+    # and show as undefined at link-time. The object library is used as a workaround.
     target_link_libraries(${test_executable} PRIVATE
-        libasdf
+        ${FYAML_LIBRARIES}
+        ${ARGP_LIBRARIES}
+        ${BZIP2_LIBRARIES}
+        ${STATGRAB_LIBRARIES}
+        ${LZ4_LIBRARIES}
+        ${ZLIB_LIBRARIES}
+        stc
+        $<TARGET_OBJECTS:libasdf_objs>
         munit
         util
+        m
```

Comment:
Yes, this is much better.

I had noticed when the tests compile it was rebuilding and linking all the libasdf modules into the tests instead of just linking to the shared lib (though even with a static lib I'm not sure why it wouldn't just link that in).

I had a feeling this ought to be fixable but didn't give it a lot of thought (with ccache it was usually not too slow).

And yes, some of the unit tests test internal APIs that are not otherwise exposed.  In the automake build I just link in the individual object files needed for those tests, but it's not very convenient.  This seems to work.